### PR TITLE
[stable/sonarqube] Fix for issue #10195 

### LINKED ISSUE #10195: [stable/sonarqube] overiding image.tag=7.4-community does not install default sonarqube plugins
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG

**Version of Helm and Kubernetes**:
helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}

minikube 

**Which chart**: stable/sonarqube

**What happened**: 
The default version of sonarqube used on this chart is 6.7.6-community. 
.net core code needs 7.4-community to scan code correctly
Overriding the image.tag in values.yaml 7.4-community does use later docker image 
HOWEVER
when its running in the pod 7.4-community does not install the default plugins for sonarqube itself. 
even tho if you run the 7.4-community image via docker itself these plugins are installed correctly. 

**The reason for this is that....** 
- the 6.7.6-community image uses sonar zip from https://binaries.sonarsource.com/Distribution/sonarqube/sonarqube-6.7.6.zip which has a structure like: 
```
/sonarqube-6.7.6/extensions/plugins$ pwd
/sonarqube-6.7.6/extensions/plugins
/sonarqube-6.7.6/extensions/plugins$ ls
README.txt
/sonarqube-6.7.6/extensions/plugins$ 

```
_BUT it does have plugins in the lib/bundled-plugins/ folder_

```
/sonarqube-6.7.6/lib/bundled-plugins$ pwd
/sonarqube-6.7.6/lib/bundled-plugins
/sonarqube-6.7.6/lib/bundled-plugins$ ls
sonar-csharp-plugin-6.5.0.3766.jar  sonar-java-plugin-4.15.0.12310.jar      sonar-php-plugin-2.11.0.2485.jar    sonar-scm-git-plugin-1.3.0.869.jar  sonar-typescript-plugin-1.1.0.1079.jar
sonar-flex-plugin-2.3.jar           sonar-javascript-plugin-3.2.0.5506.jar  sonar-python-plugin-1.8.0.1496.jar  sonar-scm-svn-plugin-1.6.0.860.jar  sonar-xml-plugin-1.4.3.1027.jar
/sonarqube-6.7.6/lib/bundled-plugins$ 
```
- the 7.4-community image uses sonar zip from https://binaries.sonarsource.com/Distribution/sonarqube/sonarqube-7.4.zip which has a structure like: 

> /sonarqube-7.4/extensions/plugins$ pwd
> /sonarqube-7.4/extensions/plugins
> /sonarqube-7.4/extensions/plugins$ ls
> README.txt                          sonar-go-plugin-1.1.0.1612.jar          sonar-kotlin-plugin-1.2.1.2009.jar   sonar-ruby-plugin-1.2.1.2009.jar        sonar-vbnet-plugin-7.7.0.7192.jar
> sonar-csharp-plugin-7.7.0.7192.jar  sonar-jacoco-plugin-1.0.1.143.jar       sonar-ldap-plugin-2.2.0.608.jar      sonar-scm-git-plugin-1.6.0.1349.jar     sonar-xml-plugin-1.5.1.1452.jar
> sonar-css-plugin-1.0.2.611.jar      sonar-java-plugin-5.8.0.15699.jar       sonar-php-plugin-2.14.0.3569.jar     sonar-scm-svn-plugin-1.9.0.1295.jar
> sonar-flex-plugin-2.4.0.1222.jar    sonar-javascript-plugin-5.0.0.6962.jar  sonar-python-plugin-1.10.0.2131.jar  sonar-typescript-plugin-1.8.0.3332.jar
> /sonarqube-7.4/extensions/plugins$ 
and nothing in the lib/bundled-plugins/ folder

SonarQube has class 'ServerPluginRepository.java' (links to both 6 and 7 branches below)
- https://github.com/SonarSource/sonarqube/blob/branch-7.0/server/sonar-server/src/main/java/org/sonar/server/plugins/ServerPluginRepository.java
- https://github.com/SonarSource/sonarqube/blob/branch-6.7/server/sonar-server/src/main/java/org/sonar/server/plugins/ServerPluginRepository.java

In this class the function 'copyBundledPlugins' copies jars from lib/bundled-plugins/ to /extensions/plugins during start up. 
... hence on 6.7.6 the plugins are not in the /extensions/plugins to start with they are copied in later
and on 7.4 they are there to start with.

PROBLEM with the Helm Chart is that it includes a mounted volume to /opt/sonarqube/extensions
via the initContainer (used to download custom extra plugins) 
k8s mounted volume is overriding the folder on base 7.4-community image and therefore that folder is empty and no plugins installed to sonarqube 
https://stackoverflow.com/questions/53862726/stop-k8s-initcontainers-volume-overwriting-container-folder

**What you expected to happen**:
7.4-community to run on k8s cluster to have the base plugins installed on startup correctly. 

**How to reproduce it** (as minimally and precisely as possible):
- override the image.tag with 7.4-community 
- install chart and wait for sonarqube to startup 
- then on sonarqube menu go to 'Quality Profiles' you will see none are installed

**Anything else we need to know**: I have done work to correct this will submit a PR for this and add PR number when submitted.

### PR DESCRIPTION
**Pull request by melissapalmer**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #10195

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/sonarqube/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: sonarqube
 description: Sonarqube is an open sourced code quality scanning tool
-version: 0.10.3
-appVersion: 6.7.6
+version: 0.10.4
```
Comment (rjkernick) [inline stable/sonarqube/Chart.yaml]: Please increment the minor version instead as this changes the underlying container version

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/sonarqube/Chart.yaml (hunk 1 above)
Comment (melissapalmer) [reply stable/sonarqube/Chart.yaml]: @rjkernick bumping this version is causing issues, as has already been done on your master branch. I have attempted to sort out but not all the sign off requirements seem to be failing. 
I have attempted to correct... but it is still complaining. 

Please can you assist, as I am unaware how to sort this out now, seems a shame to not get this work in cause of version change.

**Comment by rjkernick** [comment]
Details on the DCO here: https://github.com/helm/charts/pull/10196/checks?check_run_id=45484289
Has a couple commands that should sort it out.

**Comment by melissapalmer** [comment]
Hi @rjkernick I have attempted these commands and its just making things worse as

the commits by others are now being signed off my me and its not happy. for example below
> Junior ctadeu@gmail.com", but got "cpanato ctadeu@gmail.com".
Commit sha: f6f2a9e, Author: Carlos Tadeu Panato Junior, Committer: Melissa Palmer; Expected "Carlos Tadeu Panato Junior ctadeu@gmail.com", but got "cpanato ctadeu@gmail.com".
Commit sha: 4f796b7, Author: Carlos Tadeu Panato Junior, Committer: Melissa Palmer; Expected "Carlos Tadeu Panato Junior ctadeu@gmail.com", but got "cpanato ctadeu@gmail.com".

I started with 2 commit issues, did the commands as recommended and it then when to 44 issues
I now redid them and getting 96 issues. 

I am not sure how to sort this out now.

**Comment by melissapalmer** [comment]
Closing this PR will attempt to submit another as am unable to sort out the DCO from

**Comment by melissapalmer** [comment]
reopening as I have the same issue when opening a new PR now anyawys

**Comment by melissapalmer** [comment]
Closing this PR will attempt to submit another as am unable to sort out the DCO from

**Pull request closed without merging**